### PR TITLE
Migration/Typegoose

### PR DESCRIPTION
**Pull request by ishiasaka**

### CODE REVIEW

**Review by GolfGrab — DISMISSED** [review]
good job 👍👍

**Review by GolfGrab — DISMISSED** [review]
Excellent  👍👍

File: src/types/delete-result.type.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { Model } from 'mongoose';
+
+type DeleteFunction = typeof Model.deleteOne;
+export type DeleteResult = Awaited<ReturnType<DeleteFunction>>;
```
Comment (namirinz) [inline src/types/delete-result.type.ts:4]: What is this @hirasawaau

**Review by PraepechPrAe — COMMENTED** [review]
Look good munk

File: src/types/delete-result.type.ts (hunk 1 above)
Comment (ishiasaka) [reply src/types/delete-result.type.ts:4]: image  Error when not defined return type but the type is in mongodb dependency but can't import to this file. I created this type to get return's type of `Model.deleteOne()`

Comment (namirinz) [reply src/types/delete-result.type.ts:4]: OK

**Review by namirinz — APPROVED** [review]
LGTM

**Pull request merged**